google/resource_compute_instance.go

```diff
@@ -969,6 +1034,7 @@ func resourceComputeInstanceRead(d *schema.ResourceData, meta interface{}) error
 	d.Set("name", instance.Name)
 	d.Set("description", instance.Description)
 	d.Set("hostname", instance.Hostname)
+	d.Set("status", instance.Status)
```

Comment:
Can you check `desired_status` and only set this value if it's not `""`?

One of the problems that settable status fields have had historically is that we can't tell if the desired status is user-set or an old value from state. This should make a few interactions work predictably that could otherwise lock instances in undesirable states.

We expect the following behaviour, roughly:

* If a user never sets `desired_status`, we'll never wait for an expected status

* If a user has `desired_status` in config and imports the resource, it will not be set until an update.

* If a user unsets `desired_status` in config, Terraform will transition the value to unset (`""` represents unset) during a no-op update.

Reply:
I have updated acceptance tests to cover the 1st and 3rd cases. However, I'm not sure how to do the 2nd.